src/lib/auth.ts

```diff
@@ -66,7 +66,7 @@ export const DEVICE_API_KEY_PERMISSIONS = [
 	'resin.device_type.read?describes__device/canAccess()',
 	`resin.device.read?${matchesActor}`,
 	`resin.device.update?${matchesActor}`,
-	'resin.supervisor_release.read?should_manage__device/canAccess()',
+	'resin.supervisor_release.read?is_for__device_type/any(dt:dt/describes__device/canAccess())',
```

Comment:
this _might_ not be required (perhaps I misunderstand the type of token created/used as part of the fake-device initialization).

Reply:
Yeah I don't understand the change for this, the device should only need to read the supervisor_release that it's actually meant to be running, in this case a device can read all supervisor releases for its device type including ones that are marked as `is public = false`

Reply:
This PR addresses the case where we're not actually managed by any `supervisor_release`, so i think we'll need to access all public editions of compatible supervisors for our device-type?

Reply:
I think it makes more sense to do the `.get` itself with higher privileges rather than grant the device key these privileges as imo minimizing the power of a device key is a good approach. It's also possible that if the device at any point does `(GET /supervisor_release)[0]` that this would be a breaking change - something it definitely shouldn't rely on but I would not be shocked if it does

Reply:
I have raised the privileges for the `get` and `patch` accordingly, rather than elevating the permissions of the key itself.